### ISSUE TITLE
BaseDAO session handling

### ISSUE THREAD

**Issue by henning-gerhardt** [kitodo/kitodo-production#1983]
Our Kitodo.Production instance undergo in last couple of weeks  a lot of messages like this one

```
ERROR 2018-11-09 14:53:37,200 (Metadaten.java:calculateMetadataAndImages:741)
        de.sub.goobi.helper.exceptions.DAOException: org.hibernate.TransactionException: Already have an associated managed connection

```

or 

```
ERROR 2018-11-09 14:32:21,284 (AktuelleSchritteForm.java:SchrittDurchBenutzerUebernehmen:262)
        step couldn't get saved
de.sub.goobi.helper.exceptions.DAOException: org.hibernate.TransactionException: Already have an associated managed connection
    at de.sub.goobi.persistence.BaseDAO.rollback(BaseDAO.java:206)
    at de.sub.goobi.persistence.BaseDAO.storeObj(BaseDAO.java:172)
    at de.sub.goobi.persistence.ProzessDAO.save(ProzessDAO.java:27)
    at de.sub.goobi.forms.AktuelleSchritteForm.SchrittDurchBenutzerUebernehmen(AktuelleSchritteForm.java:259)
```

So I adjusted session and transaction handling in the BaseDAO class after my own knowledge how this should work. As no unit or functional tests are existing I'm unsure if my changes are correct or not. 

Errors appear on normal Kitodo.Production logs and in the user interface pointing to the entries in the log. On normal load until 8 / 10 concurrent logged in users only a few or no message but if concurrent user amount rise 15 until 30 messages appears at least once or twice per hour or more.

First approach was pull request #1982 but changes there introduce more errors then its resolve.

**Comment by matthias-ronge** [comment]
The way Hibernate is integrated in version 2 is inherently buggy. See also #296, #339, #339, #431, #472, #629. I'm afraid that will never change.

**Comment by henning-gerhardt** [comment]
In the past I rewrote a lot of the base dao session handling in 2.x including a lot of cleanup and simplifying. This decreased the errors a lot and a few of the mentioned issues did not appear any more. But there are still some and I did not found the reason for it. The other still existing problem in 2.x is the fact that there are 2 concurrent database access connections are used: Hibernate and Apache commons db. Some code are using the Hibernate way and some other the Apache way and some code using both :-( So this is fixing is impossible without removing one of them. This is happened on 3.x development but can not back ported to 2.x as this is a to complex code change. [My branch for this changes](https://github.com/henning-gerhardt/kitodo-production/tree/basedao_session_handling) is based on release 2.3.0 as this version is used inside SLUB. Without any kind of tests I did not reopend the pull request as this kind of changes may did not work with in different environments.
